qiskit/circuit/gate.py

```diff
@@ -48,3 +49,13 @@ def decompositions(self):
     def _define_decompositions(self):
         """ Populates self.decompositions with way to decompose this gate"""
         raise NotImplementedError("No decomposition rules defined for ", self.name)
+
+    @property
+    def representation(self):
+        """Return matrix representation if it exists else None"""
+        return self._representation
```

Comment:
should we call this `matrix_rep`, or just `matrix`? it might not be clear what "representation" means for a general gate.

Also, what happens when you try to get this property for a standard gate, such as an `x` gate? from this code it seems like it should return None, but I get an AttributeError.

Reply:
I'm fine with either. 

You can get the property like this,
```> x = qc.x(qr[0])
> x.representation 
# None
> umat = qc.unitary([[1,0],[0,1]], qr[0])
> umat.representation
array([[1.+0.j, 0.+0.j],
       [0.+0.j, 1.+0.j]], dtype=complex64)```